command/runner.go

```diff
@@ -203,3 +219,67 @@ func (r *cmdWaitReadCloser) Close() error {
 	r.closed = true
 	return nil
 }
+
+// Close the pipe and wait until the consumer program exits
+func (p pagerPipe) Close() error {
+	err := p.writer.Close()
+	<-p.cc
+	return err
+}
+
+func (p pagerPipe) Writer() io.Writer {
+	return p.writer
+}
+
+func defaultPager() string {
+	if p := os.Getenv("PAGER"); p != "" {
+		return p
+	}
+	if _, err := exec.LookPath("less"); err == nil {
+		return "less -RF"
+	}
+	if _, err := exec.LookPath("more"); err == nil {
+		return "more"
+	}
+	return ""
+}
+
+func runPager(pagerCmd string) (*pagerPipe, error) {
+	pager := pagerCmd
+	if pagerCmd == "" {
+		pager = defaultPager()
+	}
+
+	if pager == "" {
+		return nil, fmt.Errorf("no pager configured via PAGER or KUBECOLOR_PAGER environment variables")
+	}
+
+	pargs := strings.Fields(pager)
+	if _, err := exec.LookPath(pargs[0]); err != nil {
+		return nil, err
+	}
+	cmd := exec.Command(pargs[0], pargs[1:]...)
```

Comment:
Instead of using `strings.Fields`, I think the more common (and more precise) approach would be to use `sh -c "less -RF"`, granted that the user has `sh` installed (which isn't the case for Windows users, or super slim environments like wasm or some Docker images).

So what about this?

```suggestion
	cmd := exec.Command("sh", "-c", pager)
	if _, err := exec.LookPath("sh"); err != nil {
		pargs := strings.Fields(pager)
		if _, err := exec.LookPath(pargs[0]); err != nil {
			return nil, err
		}
		cmd = exec.Command(pargs[0], pargs[1:]...)
	}
```

Reply:
I'm not sure about this. I would prefer a consistent approach. I also recall that man-db doesn't execute a shell (but I'm not entirely sure).

I think the only real downside of the current approach is that you can't have arguments with spaces. If we add that option people can use their preferred shell by explicitly setting it as pager. I think that would be best.

Reply:
Sure. Can live with that reasoning